application/backend/app/services/base_weights_service.py

```diff
@@ -0,0 +1,262 @@
+# Copyright (C) 2025 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import hashlib
+import logging
+import shutil
+from pathlib import Path
+
+import requests
+
+from app.supported_models import ModelManifest
+from app.supported_models.default_models import TaskType
+from app.supported_models.supported_models import SupportedModels
+
+logger = logging.getLogger(__name__)
+
+
+class BaseWeightsService:
+    """Service for downloading and managing pretrained model weights from external archives."""
+
+    def __init__(self, data_dir: Path) -> None:
+        self._weights_cache_dir = data_dir / "models" / "pretrained_weights"
```

Comment:
```suggestion
        self._weights_cache_dir = data_dir / "pretrained_weights"
```
The root `models` folder is going to be removed, instead models will be stored inside the respective project folder (#4755).

The `pretrained_weights` dir is supposed to be directly under `data_dir`: https://github.com/open-edge-platform/training_extensions/blob/6ec7c7d6ae419eca1e2683db1e8db6def3e3939f/docs/geti_tune/models.md?plain=1#L126-L127